front/src/modules/events/hooks/useTrackPageView.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { useCreateEventMutation } from '~/generated/graphql';
+
+import { useIsTelemetryEnabled } from './useIsTelemetryEnabled';
+
+export function useTrackPageView() {
+  const location = useLocation();
+  const telemetryEnabled = useIsTelemetryEnabled();
+
+  const [createEventMutation] = useCreateEventMutation();
+
+  useEffect(() => {
```

Comment:
I would rather provide the following api:
const sendEvent = useTelemetry() 
sendEvent({ type: 'pageView', data: payload}

Reply:
Initially I had this create a similar method, but I looked at how most framework where doing and it's the most common to have a dedicated pageview method. I can do an intermediate method trackEvent yes (that will not be used anywhere in the app at this point)